File: crowdin/src/main/java/com/crowdin/platform/CrowdinInitProvider.java
package com.crowdin.platform;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.content.Context;
import android.content.pm.ProviderInfo;
import android.database.Cursor;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public class CrowdinInitProvider extends ContentProvider {

    static final String EMPTY_APPLICATION_ID_PROVIDER_AUTHORITY = "com.crowdin.platform.crowdininitprovider";

    public CrowdinInitProvider() {
    }

    public void attachInfo(Context context, ProviderInfo info) {
        checkContentProviderAuthority(info);
        super.attachInfo(context, info);
    }

    public boolean onCreate() {
        Crowdin.init(this.getContext());

        return false;
    }

    private static void checkContentProviderAuthority(@NonNull ProviderInfo info) {
        checkNotNull(info, "CrowdinInitProvider ProviderInfo cannot be null.");
        if (EMPTY_APPLICATION_ID_PROVIDER_AUTHORITY.equals(info.authority)) {
            throw new IllegalStateException("Incorrect provider authority in manifest. Most likely due to a missing applicationId variable in application's build.gradle.");
        }
    }

    @NonNull
    public static <T> T checkNotNull(T var0, Object var1) {
        if (var0 == null) {
            throw new NullPointerException(String.valueOf(var1));
        } else {
            return var0;
        }
    }

    @Nullable
    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
        return null;
    }

    @Nullable
    public String getType(Uri uri) {
        return null;
    }

    @Nullable
    public Uri insert(Uri uri, ContentValues values) {
        return null;
    }

    public int delete(Uri uri, String selection, String[] selectionArgs) {
        return 0;
    }

    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
        return 0;
    }
}

File: crowdin/src/main/java/com/crowdin/platform/CrowdinInitProvider.kt
package com.crowdin.platform

import android.content.ContentProvider
import android.content.ContentValues
import android.content.Context
import android.content.pm.ProviderInfo
import android.database.Cursor
import android.net.Uri

class CrowdinInitProvider : ContentProvider() {

    override fun attachInfo(context: Context, info: ProviderInfo) {
        checkContentProviderAuthority(info)
        super.attachInfo(context, info)
    }

    override fun onCreate(): Boolean {
        val context = this.context
        if (context != null) {
            Crowdin.init(context)
        }
        return false
    }

    override fun query(uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?, sortOrder: String?): Cursor? {
        return null
    }

    override fun getType(uri: Uri): String? {
        return null
    }

    override fun insert(uri: Uri, values: ContentValues?): Uri? {
        return null
    }

    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }

    override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }

    companion object {

        private const val EMPTY_APPLICATION_ID_PROVIDER_AUTHORITY = "com.crowdin.platform.crowdininitprovider"

        private fun checkContentProviderAuthority(info: ProviderInfo) {
            checkNotNull(info, "CrowdinInitProvider ProviderInfo cannot be null.")
            if (EMPTY_APPLICATION_ID_PROVIDER_AUTHORITY == info.authority) {
                throw IllegalStateException("Incorrect provider authority in manifest. Most likely due to a missing applicationId variable in application's build.gradle.")
            }
        }

        private fun <T> checkNotNull(var0: T?, var1: Any): T {
            return var0 ?: throw NullPointerException(var1.toString())
        }
    }
}
